### PR TITLE
Added MSE message

### PR DESCRIPTION
**Pull request by Stealeristaken**
## Proposed changes

Please include a description of the problem or feature this PR is addressing. If there is a corresponding issue, include the issue #.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: examples/python/linear_regression.py [hunk 1]
```diff
@@ -41,6 +41,6 @@ def loss_fn(w):
 throughput = num_iters / (toc - tic)
 
 print(
-    f"Loss {loss.item():.5f}, |w-w*| = {error_norm:.5f}, "
+    f"Loss {loss.item():.5f}, Mean Squared Eror : |w-w*| = {error_norm:.5f}, "
```
Comment (AtomicVar) [inline examples/python/linear_regression.py:44]: `Eror` -> `Error`?

Comment (awni) [reply examples/python/linear_regression.py:44]: It's not really an MSE (that would impliy a mean). It's really the L2 distance between $w$ and $w^*$. Maybe you could change to.

You could have `Norm of difference :` if you want to add something there for more helpful outputs.

File: examples/python/linear_regression.py [hunk 2]
```diff
@@ -41,6 +41,6 @@ def loss_fn(w):
 throughput = num_iters / (toc - tic)
 
 print(
-    f"Loss {loss.item():.5f}, Mean Squared Eror : |w-w*| = {error_norm:.5f}, "
+    f"Loss {loss.item():.5f}, L2 Distance (Norm of difference) : |w-w*| = {error_norm:.5f}, "
```
Comment (awni) [inline examples/python/linear_regression.py:44]:
```suggestion
    f"Loss {loss.item():.5f}, L2 distance: |w-w*| = {error_norm:.5f}, "
```

Let's keep it a little shorter, don't think both comments are needed

**Review by awni — APPROVED** [review]
Thanks!!

**Pull request merged**